pkg/webhook/hook.go

```diff
@@ -0,0 +1,218 @@
+package webhook
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/pkg/errors"
+	admission "k8s.io/api/admission/v1"
+	admissionregistration "k8s.io/api/admissionregistration/v1"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/cert"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+
+	v1 "github.com/percona/percona-xtradb-cluster-operator/pkg/apis/pxc/v1"
+	"github.com/percona/percona-xtradb-cluster-operator/pkg/pxctls"
+)
+
+const certPath = "/tmp/k8s-webhook-server/serving-certs/"
+
+var hookPath = "/validate-pxc-percona-com"
```

Comment:
we potentially can have validation on different objects, like backups, restores
```suggestion
var hookPath = "/PerconaXtraDBCluster"
```
can we use object `kind` as the URL?